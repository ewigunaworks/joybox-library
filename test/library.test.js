const library = require('../api/services/libraryService')

describe("Library test", () => {
  test('submit pickup schedule should return an object', async () => {
    const params = {
      title: "Adventures of Huckleberry Finn",
      authors: [
          {
              name: "Mark Twain"
          }
      ],
      genre: "History",
      editionNumber: 8157718,
      pickupSchedule: "2022-10-12 10:00:00"
    }

    const result = {
      success: true,
      status: 200,
      message: "Submit pickup schedule success.",
      data: [
        {
          title: "Adventures of Huckleberry Finn",
          authors: [
              {
                  name: "Mark Twain"
              }
          ],
          genre: "History",
          editionNumber: 8157718,
          pickupSchedule: "2022-10-12 10:00:00"
        }
      ]
    }
    await expect(library.submitPickupSchedule(params)).resolves.toMatchSnapshot(result)
  })

  test('submit pickup schedule another book should return an object', async () => {
    const params = {
      title: "Little Women",
      authors: [
          {
              "name": "Louisa May Alcott"
          }
      ],
      genre: "History",
      editionNumber: 6795866,
      pickupSchedule: "2022-10-12 11:00:00"
    }

    const result = {
      success: true,
      status: 200,
      message: "Submit pickup schedule success.",
      data: [
        {
          title: "Adventures of Huckleberry Finn",
          authors: [
              {
                  name: "Mark Twain"
              }
          ],
          genre: "History",
          editionNumber: 8157718,
          pickupSchedule: "2022-10-12 10:00:00"
        },
        {
          title: "Little Women",
          authors: [
              {
                  "name": "Louisa May Alcott"
              }
          ],
          genre: "History",
          editionNumber: 6795866,
          pickupSchedule: "2022-10-12 11:00:00"
        }
      ]
    }
    await expect(library.submitPickupSchedule(params)).resolves.toMatchSnapshot(result)
  })

  test('submit pickup schedule with same edition number should return an object with error', async () => {
    const params = {
      title: "Adventures of Huckleberry Finn",
      authors: [
          {
              name: "Mark Twain"
          }
      ],
      genre: "History",
      editionNumber: 8157718,
      pickupSchedule: "2022-10-12 10:00:00"
    }

    const result = {
      success: false,
      status: 500,
      message: `Pickup schedule with book ${params.title} and edition number ${params.editionNumber} already submitted`,
      data: []
    }
    await expect(library.submitPickupSchedule(params)).rejects.toEqual(result)
  })
})