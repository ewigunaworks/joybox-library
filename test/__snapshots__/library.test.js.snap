// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library test submit pickup schedule another book should return an object 1`] = `
{
  "data": [
    {
      "authors": [
        {
          "name": "Mark Twain",
        },
      ],
      "editionNumber": 8157718,
      "genre": "History",
      "pickupSchedule": "2022-10-12 10:00:00",
      "title": "Adventures of Huckleberry Finn",
    },
    {
      "authors": [
        {
          "name": "Louisa May Alcott",
        },
      ],
      "editionNumber": 6795866,
      "genre": "History",
      "pickupSchedule": "2022-10-12 11:00:00",
      "title": "Little Women",
    },
  ],
  "message": "Submit pickup schedule success.",
  "status": 200,
  "success": true,
}
`;

exports[`Library test submit pickup schedule should return an object 1`] = `
{
  "data": [
    {
      "authors": [
        {
          "name": "Mark Twain",
        },
      ],
      "editionNumber": 8157718,
      "genre": "History",
      "pickupSchedule": "2022-10-12 10:00:00",
      "title": "Adventures of Huckleberry Finn",
    },
  ],
  "message": "Submit pickup schedule success.",
  "status": 200,
  "success": true,
}
`;
